exist-core/src/main/java/org/exist/test/XQueryAssertions.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * eXist-db Open Source Native XML Database
+ * Copyright (C) 2001 The eXist-db Authors
+ *
+ * info@exist-db.org
+ * http://www.exist-db.org
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package org.exist.test;
+
+import com.evolvedbinary.j8fu.Either;
+import org.exist.xquery.CompiledXQuery;
+import org.exist.xquery.ErrorCodes;
+import org.exist.xquery.XPathException;
+import org.exist.xquery.value.Sequence;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class XQueryAssertions {
+    public static void assertXQStaticError(final ErrorCodes.ErrorCode expectedCode, final int line, final int column, final String expectedMessage, final Either<XPathException, CompiledXQuery> actual) {
+        assertTrue("Expected: " + expectedCode.getErrorQName() + ", but got result: " + actual.toString(), actual.isLeft());
+        final XPathException xpe = actual.left().get();
+        assertXQErrorCode(expectedCode, xpe);
+        assertXQErrorLine(line, xpe);
+        assertXQErrorColumn(column, xpe);
+        assertXQErrorMessage(expectedMessage, xpe);
+    }
+
+    public static void assertXQErrorCode(final ErrorCodes.ErrorCode expectedCode, final XPathException exception) {
+        assertEquals("Expected: " + expectedCode.getErrorQName() + ", but got: " + exception.getErrorCode().getErrorQName(),
+                expectedCode, exception.getErrorCode());
+    }
+
+    public static void assertXQErrorLine(final int expectedLine, final XPathException exception) {
+        assertEquals("Expected line to be " + expectedLine + ", but got " + exception.getLine(),
+                expectedLine, exception.getLine());
+    }
+
+    public static void assertXQErrorColumn(final int expectedColumn, final XPathException exception) {
+        assertEquals("Expected column to be " + expectedColumn + ", but got " + exception.getColumn(),
+                expectedColumn, exception.getColumn());
+    }
+
+    public static void assertXQErrorMessage(final String expectedMessage, final XPathException exception) {
+        assertEquals("Expected message to be " + expectedMessage + ", but got " + exception.getMessage(),
+                expectedMessage, exception.getMessage());
```

Comment:
Create a new method `exception.getMessageRaw()` that will only return only `XpathException.message`